pkg/controller/expectations.go

```diff
@@ -175,12 +175,18 @@ func (exp *ControlleeExpectations) isExpired() bool {
 	return clock.RealClock{}.Since(exp.timestamp) > ExpectationsTimeout
 }
 
+func getFormattedKeyFuncErrMessage(err error) error {
+	const keyFuncChangedFormat = "KeyFunc was changed, %v"
```

Comment:
Maybe just add the panic into the function, and change it to panicWithFormattedErrMsg  or something like that ?